src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationOperationsHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationOperationsHandlerMap) {
+        this.operationOperationsHandlerMap = operationOperationsHandlerMap;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationOperationsHandlerMap.get(operation);
```

Comment:
Currently, the `getOperationHandler` method returns `null` if the operation is not found in the map. Consider throwing an exception or providing a default handler to handle cases where an operation does not have a corresponding handler, to prevent potential `NullPointerException` issues.